src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,166 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final RegistrationServiceImpl REGISTRATION_SERVICE
+            = new RegistrationServiceImpl();
+    private static final StorageDaoImpl STORAGE_DAO = new StorageDaoImpl();
+    private StorageDaoImpl storageDao = new StorageDaoImpl();
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
+        Storage.people.clear();
+    }
+
+    @Test
+    void addUser_NullLogin_ThrowsException() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("qwerty1234");
+        user.setAge(17);
+        assertThrows(RegistrationException.class, () -> REGISTRATION_SERVICE.register(user));
+    }
+
+    @Test
+    void exists_Login() {
+        User user = new User();
+        user.setLogin("Alice123");
+        user.setPassword("qwerty1234");
+        user.setAge(19);
+        storageDao.add(user);
+
+        User foundUser = storageDao.get("Alice123");
+        assertNotNull(foundUser);
+        assertEquals("Alice123", foundUser.getLogin());
```

Comment:
This test seems to be testing the `get` method of `StorageDaoImpl` rather than the `register` method of `RegistrationServiceImpl`. Ensure that the test is focused on the method under test according to the task checklist.